onnxruntime/core/providers/cuda/tensor/resize_impl.cu

```diff
@@ -20,11 +20,11 @@ __device__ int NearestPixel_ROUND_PREFER_FLOOR(float x_original, bool) {
   if (x_original == static_cast<int>(x_original) + 0.5f) {
     return static_cast<int>(_Floor(x_original));
   }
-  return static_cast<int>(_Round(x_original));
+  return static_cast<int>(roundf(x_original));
 }
 
 __device__ int NearestPixel_ROUND_PREFER_CEIL(float x_original, bool) {
-  return static_cast<int>(_Round(x_original));
+  return static_cast<int>(roundf(x_original));
```

Comment:
If x_original could be negative, I think correct implementation to round to ceil is like:
```
return static_cast<int>(x_original < 0 ? rintf(x_original) : roundf(x_original));
```
If x_original is always positive, use roundf is good

Reply:
Thank you for the suggestion, let me ping you offline.

Reply:
Then why don't we just call ceil?

Reply:
I just talked to Ashwini, the original code was from her PR #2088. 

First, we are sure that roundf(-1.5) will return -2 instead of -1.  However, from the ONNX spec it's not clear whether ONNX expects -1 or -2.  So far the safest solution would be keeping the the code's behavior unchanged.  In the past release, it was always calling roundf. We may just keep it until ONNX makes it more clear or if any customer complaints.

Reply:
How about adding a comment that this assumes x_original is not negative.

Reply:
Ashwini said we couldn't make such an assumption.

Reply:
Nearest neighbor in image interpolation is based on pixel coordinates. For image resize, I think the assumption is fine (value of coordinates >= 0).

Reply:
@edgchen1 what do you think?

Reply:
sounds good to me